src/core/ai/session/chat-session.ts

```diff
@@ -241,40 +242,7 @@ export class ChatSession {
             `Running session ${this.id} with input: ${input}, imageDataInput: ${imageDataInput}, fileDataInput: ${fileDataInput}`
         );
 
-        // Validate file format support if file data is provided
-        if (fileDataInput) {
-            const currentConfig = this.services.stateManager.getLLMConfig(this.id);
-            const { validateFileForLLM } = await import('../../ai/llm/validation.js');
-
-            const config = {
-                provider: currentConfig.provider,
-                ...(typeof currentConfig.model === 'string' && { model: currentConfig.model }),
-            };
-            const validation = validateFileForLLM(config, fileDataInput.mimeType);
-
-            if (!validation.isSupported) {
-                const error = new Error(
-                    validation.error || 'File type not supported by current LLM'
-                );
-                logger.error(`File validation failed in session ${this.id}: ${error.message}`);
-
-                // Emit event for external handlers
-                this.eventBus.emit('saiki:fileValidationError', {
-                    sessionId: this.id,
-                    mimeType: fileDataInput.mimeType,
-                    fileType: validation.fileType,
-                    provider: currentConfig.provider,
-                    model: currentConfig.model,
-                    error: error.message,
-                });
-
-                throw error;
-            }
-
-            logger.debug(
-                `File validation passed for ${validation.fileType} file in session ${this.id}`
-            );
-        }
+        // Note: Input validation is now handled at the API layer before reaching this method
```

Comment:
we still need to validate again here, because not all flows will go through API layer (SDK users)

we can re-use the validateInputForLLM.. func here and emit unsupported input events  in case of the validation errors. need to update event bus also to add this as a known event